utils/outputwriter/markdowntable.go

```diff
@@ -111,22 +141,51 @@ func (t *MarkdownTableBuilder) Build() string {
 		return ""
 	}
 	var tableBuilder strings.Builder
+	// Calculate Hidden columns
+	for c := range t.columns {
+		// Reset shouldHideColumn flag
+		t.columns[c].shouldHideColumn = t.columns[c].OmitEmpty
+	}
+	for _, row := range t.rows {
+		for c, cell := range row {
+			// In table, empty cell = cell with no values = cell with one empty value
+			t.columns[c].shouldHideColumn = t.columns[c].shouldHideColumn && (len(cell) == 0 || (len(cell) == 1 && cell[0] == ""))
+		}
+	}
 	// Header
-	for c, column := range t.columns {
-		if c == 0 {
-			tableBuilder.WriteString(fmt.Sprintf(cellFirstCellPlaceholder, column.Name))
+	actualColumnCount := 0
```

Comment:
not critical, but you can simple have a bool var 'isFirstCol = true' and check it in the condition. when you enter the first column action just change it to false. I cannot see a reason to keep counting after the first col